### PR TITLE
chore(e2e): slow down requests and cleanup

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

* added delay between requests because it seems there might be some overload of envoy - curl return 28 which means connection timeout
* added cleanup of resources between VO test
* added more retries on cluster setup for multizone
* 
- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: test/framework/client/collect.go [hunk 1]
```diff
@@ -102,6 +113,12 @@ func WithoutRetries() CollectResponsesOptsFn {
 	}
 }
 
+func WithoutConcurrentRequestDelay() CollectResponsesOptsFn {
+	return func(opts *CollectResponsesOpts) {
+		opts.withoutConcurrentRequestDelay = true
```
Comment (lahabana) [inline test/framework/client/collect.go:118]: Why not just use 0?

File: test/framework/client/collect.go [hunk 2]
```diff
@@ -84,12 +88,19 @@ func WithCACert(cacert string) CollectResponsesOptsFn {
 	}
 }
 
-func WithMaxConcurrentReqeusts(maxConcurrentRequests int) CollectResponsesOptsFn {
+func WithMaxConcurrentRequests(maxConcurrentRequests int) CollectResponsesOptsFn {
```
Comment (lahabana) [inline test/framework/client/collect.go:91]: Unit please :)

File: test/framework/client/collect.go [hunk 3]
```diff
@@ -25,33 +26,36 @@ import (
...
 	return CollectResponsesOpts{
-		numberOfRequests:      10,
-		maxConcurrentRequests: 10,
-		maxTime:               5,
-		verbose:               false,
-		Method:                "GET",
-		Headers:               map[string]string{},
-		ShellEscaped:          utils.ShellEscape,
+		numberOfRequests:          10,
+		maxConcurrentRequests:     10,
```
Comment (lahabana) [inline test/framework/client/collect.go:52]: Should we reduce this? like 5 or something like this?

Comment (lukidzi) [reply test/framework/client/collect.go:52]: We can do it because it produces a bit of load on Envoy.

File: .github/workflows/_e2e.yaml [hunk 4]
```diff
@@ -130,6 +130,9 @@ jobs:
             export GINKGO_E2E_LABEL_FILTERS="job-${{ matrix.parallelRunnerId }}"
           fi
           env
+          if [[ "${{ env.E2E_PARAM_TARGET }}" == "multizone" ]]; then
```
Comment (lukidzi) [inline .github/workflows/_e2e.yaml:133]:
```
Summarizing 3 Failures:
  [FAIL] [SynchronizedBeforeSuite] 
  github.com/kumahq/kuma/test/framework/ginkgo.go:46
  [FAIL] [SynchronizedBeforeSuite] 
  github.com/kumahq/kuma/test/framework/envs/multizone/env.go:106
  [PANICKED!] [DeferCleanup (Suite)] 
```

there was a message:
```
          Err: <retry.MaxRetriesExceeded>{
              Description: "get default mesh",
              MaxRetries: 30,
          },```

want to see if that helps

Comment (lukidzi) [reply .github/workflows/_e2e.yaml:133]: seems to help

**Review by Automaat — APPROVED** [review]

**Pull request merged**